Lib/test/test_typing.py

```diff
@@ -4228,6 +4246,90 @@ def test_get_type_hints(self):
             {'a': typing.Optional[int], 'b': int}
         )
 
+    def test_get_type_hints_generic(self):
+        self.assertEqual(
+            get_type_hints(BarGeneric[int].__origin__), 
+            {'a': typing.Optional[T], 'b': int}
+        )
+
+    def test_generic_inheritance(self):
+        class A(TypedDict, Generic[T]):
+            a: T
+
+        self.assertEqual(A.__bases__, (Generic, dict,))
+        self.assertEqual(A.__parameters__, (T,))
+        self.assertEqual(A[str].__parameters__, ())
+        self.assertEqual(A[str].__args__, (str,))
+
+        class B(A[KT], total=False):
+            b: KT
+
+        self.assertEqual(B.__bases__, (Generic, dict,))
+        self.assertEqual(B.__parameters__, (KT, ))
+        self.assertEqual(B.__total__, False)
+        self.assertEqual(B.__optional_keys__, frozenset(['b']))
+        self.assertEqual(B.__required_keys__, frozenset(['a']))
+
+        self.assertEqual(B[str].__parameters__, ())
+        self.assertEqual(B[str].__args__, (str,))
+        self.assertEqual(B[str].__origin__, B)
+
+        class C(B[int]):
+            c: int
+
+        self.assertEqual(C.__bases__, (Generic, dict,))
+        self.assertEqual(C.__parameters__, ())
+        self.assertEqual(C.__total__, True)
+        self.assertEqual(C.__optional_keys__, frozenset(['b']))
+        self.assertEqual(C.__required_keys__, frozenset(['a', 'c']))
+        assert C.__annotations__ == {
```

Comment:
Yeah it would have been nice. I tried a previous patch that tried to [resolve the types](https://github.com/python/cpython/compare/3.9...sransara:py-generic-typeddict).

But since the following code with normal generic classes doesn't resolve types:
```
from typing import *
T = TypeVar("T")
class A(Generic[T]):
    a: T
class B(A[int]):
    b: str
print(get_type_hints(B))
# {'a': ~T, 'b': <class 'str'>}
``` 

And after seeing the [discussion here](https://github.com/python/typing/issues/629#issuecomment-493977366) I decided to keep the implementation simple and not try to resolve the type with this patch.

Reply:
Yeah I think you're right, the current behavior is consistent with normal generic classes.